urnx/parse.go

```diff
@@ -0,0 +1,29 @@
+package urnx
+
+import (
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+func Parse(urn string) (*URN, error) {
+	conv := strings.Split(urn, ":")
+
+	if conv[0] != PREFIX {
+		return nil, ErrInvalidURNPrefix
+	}
+
+	if len(conv) != URNLENGTH {
+		return nil, ErrInvalidURN
+	}
+
+	u := &URN{
+		Prefix:       conv[0],
+		Namespace:    conv[1],
```

Comment:
For completeness, it might be good to validate that `Namespace` conforms to the URN grammar, which is 0-30 alphanumeric characters: https://www.rfc-editor.org/rfc/rfc8141.html#section-2

Additionally, namespace identifiers are case insensitive. For ease of equality checking it might be good to make them lowercase when parsing the URN.

Reply:
Added validation for namespaces and pushed everything to `lowercase`